### PR TITLE
Make catalog import asynchronous

### LINKED ISSUE #961: Make catalog_import API endpoint async
The Dev-VM often throws an exception when triggering a catalog import.

```
$ src/lizards-and-pumpkins/bin/clearFileStorages.php
$ /vagrant/sample-project/build/buildLizardsAndPumpkinsSnippets.sh /vagrant/sample-project
Fatal error: Uncaught GuzzleHttp\Exception\ServerException: Server error: `PUT http://demo.lizardsandpumpkins.com.loc/fr/api/catalog_import/` resulted in a `504 Gateway Time-out` response:
<html>
<head><title>504 Gateway Time-out</title></head>
```

GIven that all it does it plop a command on the queue I can't see a reason why that is taking so long (without looking into the code), but I'm pretty sure we can make that problem go away.

### PR DESCRIPTION
**Pull request by Vinai**
Instead of processing the import file in the API request handler, a new Command `ImportCatalogCommand` is created.  
This command validates the import file exists.  
The command handler then creates a `CatalogImportWasTriggeredDomainEvent`.  
The `CatalogImportWasTriggeredDomainEventHandler`  then triggers the actual import.

Closes #961

### CODE REVIEW

File: src/Util/Factory/CommonFactory.php [hunk 1]
```diff
@@ -992,6 +995,11 @@ public function createShutdownWorkerCommandHandler(Message $message) : CommandHa
             $this->getMasterFactory()->getLogger()
         );
     }
+    
+    public function createImportCatalogCommandHandler(Message $message) : CommandHandler
```
Comment (tim-bezhashvyly) [inline src/Util/Factory/CommonFactory.php]: Space in front of a colon.

File: src/Import/CatalogImportWasTriggeredDomainEventHandler.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    private $catalogImport;
+
+    /**
+     * @var CatalogImportWasTriggeredDomainEvent
+     */
+    private $catalogImportWasTriggeredEvent;
+
+    public function __construct(
+        CatalogImport $catalogImport,
+        Message $eventMessage
```
Comment (Vinai) [inline src/Import/CatalogImportWasTriggeredDomainEventHandler.php]: Fixed.

File: src/Import/Exception/NoImportCatalogCommandMessageException.php [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace LizardsAndPumpkins\Import\Exception;
+
+class NoImportCatalogCommandMessageException extends \OutOfBoundsException
```
Comment (Vinai) [inline src/Import/Exception/NoImportCatalogCommandMessageException.php]: Renamed.

File: src/Import/ImportCatalogCommand.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        $metadata = ['data_version' => (string) $this->dataVersion];
+
+        return Message::withCurrentTime(self::CODE, $payload, $metadata);
+    }
+
+    /**
+     * @param Message $message
+     * @return static
+     */
+    public static function fromMessage(Message $message)
```
Comment (Vinai) [inline src/Import/ImportCatalogCommand.php]: Not required, but since it is correct, why not?

File: src/Import/ImportCatalogCommand.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        }
+    }
+
+    private function validateFileIsReadable(string $catalogDataFile)
+    {
+        if (!is_readable($catalogDataFile)) {
+            $message = sprintf('Catalog import file "%s" is not readable', $catalogDataFile);
+            throw new CatalogImportFileNotReadableException($message);
+        }
+    }
```
Comment (Vinai) [inline src/Import/ImportCatalogCommand.php]: Added validation case.

File: src/Util/Factory/CommonFactory.php (hunk 1 above)
Comment (Vinai) [inline src/Util/Factory/CommonFactory.php]: Fixed.

**Comment by Vinai** [comment]
Ready for next review.

**Pull request merged**